packages/sdk/client-services/src/packlets/vault/vault-resource-manager.ts

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { asyncTimeout, Trigger } from '@dxos/async';
+import { log } from '@dxos/log';
+
+import { ClientServicesHost } from '../services';
+
+const LOCK_KEY = 'DXOS_COMPATIBILITY_LOCK';
+
+enum Message {
+  ACQUIRING = 'acquiring'
+}
+
+export class VaultResourceManager {
+  private readonly _broadcastChannel = new BroadcastChannel('vault-resource-manager');
```

Comment:
extract to const

Reply:
I'm unsure if this is a good idea since the message handler for is it assigned in the constructor?